src/nvim/edit.c

```diff
@@ -2666,6 +2666,37 @@ void ins_compl_show_pum(void)
   curwin->w_cursor.col = compl_col;
   pum_display(compl_match_array, compl_match_arraysize, cur, array_changed);
   curwin->w_cursor.col = col;
+
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+  if (cur < 0) {
+    tv_dict_add_dict(dict, S_LEN("completeitem"), tv_dict_alloc());
+  } else {
+    dict_T *item = tv_dict_alloc();
+    tv_dict_add_str(
+        item, S_LEN("word"),
+        (const char *)EMPTY_IF_NULL(compl_curr_match->cp_str));
+    tv_dict_add_str(
+        item, S_LEN("abbr"),
+        (const char *)EMPTY_IF_NULL(compl_curr_match->cp_text[CPT_ABBR]));
+    tv_dict_add_str(
+        item, S_LEN("menu"),
+        (const char *)EMPTY_IF_NULL(compl_curr_match->cp_text[CPT_MENU]));
+    tv_dict_add_str(
+        item, S_LEN("kind"),
+        (const char *)EMPTY_IF_NULL(compl_curr_match->cp_text[CPT_KIND]));
+    tv_dict_add_str(
+        item, S_LEN("info"),
+        (const char *)EMPTY_IF_NULL(compl_curr_match->cp_text[CPT_INFO]));
```

Comment:
This is the same code as in `ins_compl_insert`, so the common code could be factored into a new function `ins_compl_dict_alloc(compl_curr_match->cp_text)`.

Reply:
done